370-auth-ui/oidc.md

```diff
@@ -0,0 +1,452 @@
+
+### Authentication Request
+
+The following sessions list out the request parameters that Skygear Auth supports.
+
+#### scope
+
+The value cloud be:
+
+- `["openid"]`: First party web app
+- `["openid", "offline_access"]`: Generic OIDC RP or Native app
+
+#### response_type
+
+- `code`: Using Authorization Code Flow
+- `none`: Use in [First party web app](#case-2-first-party-web-app-traditional-web-app--server-side-rendering-app--single-page-app), Idp session is used when requesting api
+
+#### client_id
+
+The value must be a client id which configured in App Config.
+
+#### redirect_uri
+
+No difference from the spec.
+
+#### state
+
+No difference from the spec.
+
+#### nonce
+
+No difference from the spec.
+
+#### display
+
+It is not supported. The only supported display mode is `page`.
+
+#### prompt
+
+- not specified: Prompt user for authentication or authorization, depends on Idp session state.
+- `none`: No difference from the spec. For requesting new access token when token expire in SPA.
+
+#### max_age
+
+It is not supported.
+
+#### ui_locales
+
+No difference from the spec.
+
+#### id_token_hint
+
+No difference from the spec, for `prompt=none` case.
+
+#### login_hint
+
+It is not supported.
+
+#### acr_values
+
+It is not supported.
+
+#### code_challenge_method
+
+Only `S256` is supported. `plain` is not supported. The reason is the existing implementation of SSO is using `S256` already. Since it defaults to plain, it is required and the value must be `S256`.
+
+#### code_challenge
+
+No difference from the spec.
+
+#### Example of valid Authentication Request
+
+`response_type=code&scope=openid&client_id=CLIENT_ID&redirect_uri=REDIRECT_URI&code_challenge_method=S256&code_challenge=CODE_CHALLENGE&ui_locales=en`
+
+### Token Request
+
+The following sessions list out the request parameters and their difference from the spec.
+
+#### grant_type
+
+Either `authorization_code` or `refresh_token`
+
+#### code
+
+No difference from the spec.
+
+#### redirect_uri
+
+No difference from the spec.
+
+#### client_id
+
+It is always required.
+
+#### code_verifier
+
+No difference from the spec.
+
+#### Example of valid Token Request
+
+`grant_type=authorization_code&code=AUTHORIZATION_CODE&redirect_uri=REDIRECT_URI&client_id=API_KEY&code_verifier=CODE_VERIFIER`
+
+### Token Response
+
+#### id_token
+
+No difference from the spec.
+
+#### token_type
+
+It is always the value `bearer`.
+
+#### access_token
+
+No difference from the spec.
+
+#### expires_in
+
+No difference from the spec.
+
+#### refresh_token
+
+No difference from the spec. Exist only if authorized scopes have `offline_access`.
+
+#### scope
+
+It is always absent.
+
+#### Example of Token Response
+
+```json
+{
+  "token_type": "bearer",
+  "access_token": "access.token",
+  "refresh_token": "refresh.token"
+}
+```
+
+## Authorization Server Metadata Document
```

Comment:
I think we should also document every endpoints (e.g. userinfo, jwks_uri), and note down which fields belongs to which document (I think both document can have same fields?)